File: parking-domain/src/main/java/com/chatchatabc/parking/impl/infra/schedule/user/job/UserLoginSendSMSOTPJob.java
package com.chatchatabc.parking.impl.infra.schedule.user.job;

import org.quartz.Job;
import org.quartz.JobDataMap;
import org.quartz.JobExecutionContext;

public class UserLoginSendSMSOTPJob implements Job {

    /**
     * Execute the job of sending sms otp
     *
     * @param context the job execution context
     */
    @Override
    public void execute(JobExecutionContext context) {
        JobDataMap dataMap = context.getJobDetail().getJobDataMap();
        String phone = dataMap.getString("phone");
        String otp = dataMap.getString("otp");

        // TODO: Send text message with technology. OTP will print for now
        System.out.println("Phone: " + phone + ", OTP: " + otp);
    }
}

File: parking-api/src/main/kotlin/com/chatchatabc/parking/api/application/schedule/user/job/UserLoginSendSMSOTPJob.kt
package com.chatchatabc.parking.api.application.schedule.user.job

import org.quartz.Job
import org.quartz.JobExecutionContext

class UserLoginSendSMSOTPJob : Job {

    /**
     * Execute the job of sending sms otp
     */
    override fun execute(context: JobExecutionContext) {
        val dataMap = context.jobDetail.jobDataMap
        val phone = dataMap.getString("phone")
        val otp = dataMap.getString("otp")

        // TODO: Send text message with technology. OTP will print for now
        println("Phone: $phone, OTP: $otp")
    }
}